apps/site/assets/ts/stop/components/StopPageRedesign.tsx

```diff
@@ -10,28 +10,50 @@ import { useSchedulesByStop } from "../../hooks/useSchedules";
 import { useAlertsByRoute, useAlertsByStop } from "../../hooks/useAlerts";
 import DeparturesAndMap from "./DeparturesAndMap";
 import { isCurrentAlert, routeWideAlerts } from "../../models/alert";
+import { FetchStatus } from "../../helpers/use-fetch";
 
 const StopPageRedesign = ({
   stopId
 }: {
   stopId: string;
 }): ReactElement<HTMLElement> => {
-  const stop = useStop(stopId);
-  const routesWithPolylines = useRoutesByStop(stopId);
+  const stopResult = useStop(stopId);
+  const routesWithPolylinesResult = useRoutesByStop(stopId);
+  const schedulesResult = useSchedulesByStop(stopId);
+  const alertsForStopResult = useAlertsByStop(stopId);
+  const alertsForRoutesResult = useAlertsByRoute(
+    routesWithPolylinesResult.status === FetchStatus.Data
+      ? routesWithPolylinesResult.data?.map(r => r.id) || []
+      : []
+  );
 
-  const routes = routesWithPolylines
-    ? routesWithPolylines.map(rwp => omit(rwp, "polylines"))
-    : [];
-  const routeIds = routes.map(r => r.id);
+  if (
+    [
+      stopResult.status,
+      routesWithPolylinesResult.status,
+      schedulesResult.status
+    ].includes(FetchStatus.Error)
+  ) {
+    return <p>Page could not be loaded. Please try refreshing the page.</p>;
+  }
 
-  // TODO maybe move to the StopDeparturesPage (or keep it here for loading indicator)
-  const schedules = useSchedulesByStop(stopId);
-  const alertsForStop = useAlertsByStop(stopId);
-  const alertsForRoutes = useAlertsByRoute(routeIds);
-  //
+  // Return loading indicator while waiting on data fetch
+  if (
+    !stopResult.data ||
+    !routesWithPolylinesResult.data ||
+    !schedulesResult.data ||
+    !alertsForRoutesResult.data ||
+    !alertsForStopResult.data
```

Comment:
Ah thanks for noticing that! I've updated to do the opposite - since we are checking that data exists here, we can remove the fallback on lines 55-56. I think we would want to preserve the loading state while alerts load based on [this test case](https://github.com/mbta/dotcom/blob/b323b669f9f71131e77ee978117b03cdef2477af/apps/site/assets/ts/stop/__tests__/StopPageRedesignTest.tsx#L70)